subprojects/core/src/main/java/org/gradle/api/execution/internal/ExecuteTaskBuildOperationDetails.java

```diff
@@ -53,4 +57,14 @@ public long getTaskId() {
         return task.getTaskIdentity().type;
     }
 
+    @Override
+    public Object getCustomOperationTraceSerializableModel() {
+        Map<String, Object> map = new HashMap<String, Object>();
```

Comment:
:o: I'd probably use `ImmutableMap.Builder` here, if it doesn't impact performance.